crypto/crypto.go

```diff
@@ -445,39 +450,74 @@ func VerifyLogicSig(lsig types.LogicSig, sender types.Address) (result bool) {
 		return false
 	}
 
-	result = false
 	toBeSigned := programToSign(lsig.Logic)
-	// logic sig, compare hashes
-	if !hasSig && !hasMsig {
-		result = types.Digest(sha512.Sum512_256(toBeSigned)) == types.Digest(sender)
-		return
-	}
```

Comment:
why this did go? Shouldn't non-delegated logicsig continue to be prechecked?

Reply:
The only purpose of this check is to verify that the sender is the LogicSig's hashed program address. This check is no longer needed, since we want to support signing txns for accounts rekeyed to LogicSigs.